### PR TITLE
feature/SKED-7749: added appointments

### PR DESCRIPTION
**Pull request by Pramuspl**
> See JIRA https://skedify.atlassian.net/browse/SKED-7749

### CODE REVIEW

File: types/appointment.d.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    contact_id: number
+    customer_id: number
+    subject_id: number
+    office_id: number
+    meeting_type: string
+    first_possibility_id: number
+    notes?: string
+    outcome_remark?: string
+    trailing_buffer_time: number
+    metadata: any[]
```
Comment (Jarrku) [inline types/appointment.d.ts:13]: Prefer unknown :)

**Review by Jarrku — APPROVED** [review]

File: types/appointment.d.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+declare module 'skedify-sdk' {
+  export interface Appointment extends Omit<BaseFields, 'enterprise_id'> {
+    state: string
+    contact_id: number
+    customer_id: number
+    subject_id: number
+    office_id: number
```
Comment (Jarrku) [inline types/appointment.d.ts:7]:
```suggestion
    contact_id: string
    customer_id: string
    subject_id: string
    office_id: string
```

File: types/appointment.d.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    meeting_type: string
+    first_possibility_id: number
```
Comment (Jarrku) [inline types/appointment.d.ts:9]:
```suggestion
    first_possibility_id: string
```

File: types/appointment.d.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    initiated_by_id: number
+    initiated_by_type: string
+    updated_at: string
+    updated_by_id: number
+    updated_by_type: string
+  }
+
+  export interface Possibility
+    extends Omit<BaseFields, 'enterprise_id' | 'external_id'> {
+    appointment_id: number
```
Comment (Jarrku) [inline types/appointment.d.ts:33]:
```suggestion
    appointment_id: string
```

File: types/appointment.d.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+declare module 'skedify-sdk' {
+  export interface Appointment extends Omit<BaseFields, 'enterprise_id'> {
+    state: string
+    contact_id: number
```
Comment (BenSwennen) [inline types/appointment.d.ts:4]: Can you update the FKs to string? SDK Parses these as string

https://github.com/skedify/sdk-js/blob/6a10b3acf26dac6655f8c4a2d53488f7400ede63/src/decorators/withResources/normalizeResponse.js#L27

**Review by Jarrku — CHANGES_REQUESTED** [review]

File: types/appointment.d.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+declare module 'skedify-sdk' {
+  export type AppointmentState =
+    | 'incoming'
+    | 'outgoing'
+    | 'completed'
+    | 'cancelled'
+    | 'accepted'
+    | 'occurred'
+
+  export interface Appointment extends Omit<BaseFields, 'enterprise_id'> {
```
Comment (kendepelchin) [inline types/appointment.d.ts:10]: Q: Why is uuid not added to the Appointment?

Q: Why is listing_uuid not added to the Appointment?

File: types/appointment.d.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    video_url_for_customer: string;
+    video_test_url_for_contact: string;
+    video_test_url_for_customer: string;
+    cancellation_reason_id?: string;
+    cancelled_by_id?: string;
+    cancelled_by_type?: string;
+    initiated_from: string;
+    initiated_by_id: string;
+    initiated_by_type: string;
+    updated_at: string;
```
Comment (kendepelchin) [inline types/appointment.d.ts:34]: Shouldn't `created_at` be added as well?

File: types/appointment.d.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+  export interface Possibility
+    extends Omit<BaseFields, 'enterprise_id' | 'external_id'> {
+    appointment_id: number;
+    start: string;
+    end: string;
+    timezone: string;
+    duration: number;
+    is_accepted: boolean;
+    next_possibility_id: null;
```
Comment (kendepelchin) [inline types/appointment.d.ts:47]: Q: According to the API this value is always an integer. Shouldn't this reflect this as well?

File: types/appointment.d.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    updated_at: string;
+    updated_by_id: string;
+    updated_by_type: string;
+  }
+
+  export interface Possibility
+    extends Omit<BaseFields, 'enterprise_id' | 'external_id'> {
+    appointment_id: number;
+    start: string;
+    end: string;
```
Comment (kendepelchin) [inline types/appointment.d.ts:43]: These dates are probably parsed as dates?

File: types/appointment.d.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    timezone: string;
+    duration: number;
+    is_accepted: boolean;
+    next_possibility_id: null;
+    created_by_id: number;
+    created_by_type: string;
+    updated_by_id: number;
+    updated_by_type: string;
+    updated_at: string;
```
Comment (kendepelchin) [inline types/appointment.d.ts:52]: Same Q for `created_at`

File: types/customer.d.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    language: string;
+    phone_number?: string;
+    profile_picture?: string;
+    timezone: string;
+    customer_number?: string;
+    company?: string;
+    notes?: string;
+    is_existing: boolean;
+    updated_at: string;
+    location?: string;
```
Comment (kendepelchin) [inline types/customer.d.ts:21]: Same Q for uuid

File: types/customer.d.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    date_of_birth?: string;
+    language: string;
+    phone_number?: string;
+    profile_picture?: string;
+    timezone: string;
+    customer_number?: string;
+    company?: string;
+    notes?: string;
+    is_existing: boolean;
+    updated_at: string;
```
Comment (kendepelchin) [inline types/customer.d.ts:20]: Same Q for `created_at`

**Review by kendepelchin — CHANGES_REQUESTED** [review]
I've added some Q's :pray:

File: types/appointment.d.ts (hunk 6 above)
Comment (BenSwennen) [reply types/appointment.d.ts:10]: Added those

File: types/appointment.d.ts (hunk 7 above)
Comment (BenSwennen) [reply types/appointment.d.ts:34]: Appointment extends from BaseFields, which types `id`, `created_at` and `external_id`

File: types/appointment.d.ts (hunk 9 above)
Comment (BenSwennen) [reply types/appointment.d.ts:43]: SDK returns them as strings, not parsed yet:

```
end: "2021-08-09T17:00:00+02:00"
start: "2021-08-09T16:30:00+02:00"
```

File: types/appointment.d.ts (hunk 8 above)
Comment (BenSwennen) [reply types/appointment.d.ts:47]: I haven't had a case where I had an actual next possibility, but will add `number | null`

File: types/appointment.d.ts (hunk 10 above)
Comment (BenSwennen) [reply types/appointment.d.ts:52]: See BaseFields in shared.d.ts

File: types/customer.d.ts (hunk 12 above)
Comment (BenSwennen) [reply types/customer.d.ts:20]: See BaseFields in shared.d.ts

File: types/appointment.d.ts (hunk 9 above)
Comment (kendepelchin) [reply types/appointment.d.ts:45]: Yes, I meant these are parsed where used in the web admin as dates; The api always returns these as strings

**Review by kendepelchin — APPROVED** [review]

**Pull request merged**